File: estructura_datos/ordenación/selection_sort/ejemploV1.py
def selection_sort(l):
    last_position = len(l) - 1
    initial_value = 0

    while initial_value < last_position:
        position_min = initial_value

        for i in range(initial_value, last_position):
            if l[position_min] > l[i + 1]:
                position_min = i + 1

        aux = l[position_min]
        l[position_min] = l[initial_value]
        l[initial_value] = aux

        initial_value += 1

    return l


def run():
    print(selection_sort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]))
    print(selection_sort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]))


if __name__ == "__main__":
    run()

File: estructura_datos/ordenación/selection_sort/ejemploV1.js
const selectionSort = (a) => {
  const lastPosition = a.length - 1;
  let initialValue = 0;

  while (initialValue < lastPosition) {
    let positionMin = initialValue;

    for (let i = initialValue; i < lastPosition; i++) {
      if (a[positionMin] > a[i + 1]) {
        positionMin = i + 1;
      }
    }
    let aux = a[positionMin];
    a[positionMin] = a[initialValue];
    a[initialValue] = aux;

    initialValue++;
  }

  return a;
};

console.log(selectionSort([2, 7, 1, 5, 0, 4, 3, 8, -1, 2, 4]));
console.log(selectionSort([2, 7, 1, 4, 3, 5, 0, 8, 2, -1, 2]));
console.log(selectionSort([7, 2, 1, 4, 0, 5, 6, 8, 2]));
